xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/PreviewAction.java

```diff
@@ -125,15 +131,19 @@ public String render(XWikiContext context) throws XWikiException
 
         // The current user editing the document should be displayed as author and creator (if the edited document is
         // new) when the edited document is previewed.
-        editedDocument.setAuthorReference(context.getUserReference());
+        UserReference currentUserReference = this.currentUserResolver.resolve(CurrentUserReference.INSTANCE);
+        editedDocument.getAuthors().setOriginalMetadataAuthor(currentUserReference);
         if (editedDocument.isNew()) {
-            editedDocument.setCreatorReference(context.getUserReference());
+            editedDocument.getAuthors().setCreator(currentUserReference);
         }
 
         // Make sure the current user doesn't use the programming rights of the previous content author (by editing a
         // document saved with programming rights, changing it and then previewing it). Also make sure the code
         // requiring programming rights is executed in preview mode if the current user has programming rights.
-        editedDocument.setContentAuthorReference(context.getUserReference());
+        context.getRequest().getEffectiveAuthor().ifPresent(editedDocument.getAuthors()::setEffectiveMetadataAuthor);
```

Comment:
Sounds like the comment above should be moved below, and a new comment added for this call no? Also just a minor comment but might worth using a local variable for `editedDocument.getAuthors()` to simplify a bit the reading here.